src/main/java/hudson/plugins/templateproject/ProxyBuilder.java

```diff
@@ -104,18 +99,25 @@ public FormValidation doCheckProjectName(@AncestorInPath AccessControlled anc, @
 	}
 
 	@Override
-	public boolean perform(AbstractBuild<?, ?> build, Launcher launcher,
-			BuildListener listener) throws InterruptedException, IOException {
-		for (Builder builder: getProjectBuilders(build)) {
-			AbstractProject p = TemplateUtils.getProject(getProjectName(), build);
-			listener.getLogger().println("[TemplateProject] Starting builders from: " + HyperlinkNote.encodeTo('/'+ p.getUrl(), p.getFullDisplayName()));
-			if (!builder.perform(build, launcher, listener)) {
-				listener.getLogger().println("[TemplateProject] FAILED performing builders from: '" + p.getFullDisplayName() + "'");
-				return false;
+	public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
+		AbstractProject project = TemplateUtils.getInstance().getProject(getProjectName(), build);
+
+		if(project != null) {
+			for (Builder builder: getProjectBuilders(build)) {
+				listener.getLogger().println("[TemplateProject] Starting builders from: " + HyperlinkNote.encodeTo('/' + project.getUrl(), project.getFullDisplayName()));
+				if (!builder.perform(build, launcher, listener)) {
+					listener.getLogger().println("[TemplateProject] FAILED performing builders from: '" + project.getFullDisplayName() + "'");
+					return false;
+				}
+				listener.getLogger().println("[TemplateProject] Successfully performed builders from: '" + project.getFullDisplayName() + "'");
 			}
-			listener.getLogger().println("[TemplateProject] Successfully performed builders from: '" + p.getFullDisplayName() + "'");
+
+			return true;
+
+		} else {
+			listener.getLogger().printf("[TemplateProject] FAILED builders loading, couldn't find project: '%s' %n", getProjectName());
```

Comment:
It may break some use-cases, but I also vote for having it as a default behavior